api/src/main/java/org/pmiops/workbench/db/dao/UserRecentResourceServiceImpl.java

```diff
@@ -84,6 +88,23 @@ public void updateCohortEntry(long workspaceId, long userId, long cohortId,
     getDao().save(resource);
   }
 
+  @Override
+  public void updateConceptSetEntry(long workspaceId, long userId, long conceptSetId,
+                                   Timestamp lastAccessDateTime) {
+    ConceptSet conceptSet = conceptSetDao.findOne(conceptSetId);
+    UserRecentResource resource = getDao().findByUserIdAndWorkspaceIdAndConceptSet(userId, workspaceId, conceptSet);
+    if (resource == null) {
+      handleUserLimit(userId);
+      resource = new UserRecentResource();
+      resource.setUserId(userId);
+      resource.setWorkspaceId(workspaceId);
+      resource.setConceptSet(conceptSet);
+      resource.setNotebookName(null);
```

Comment:
Not a request - just a comment for posterity. I wish our dao objects had a builder pattern constructor.